src/02-room-details.js

```diff
@@ -25,7 +25,25 @@ const exampleRoomData = require("../data/rooms");
  *  getRoomByDinosaurName(dinosaurs, rooms, "Pterodactyl");
  *  //> "Dinosaur with name 'Pterodactyl' cannot be found."
  */
-function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
+ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {
+  let objId = [];
+  let dinoRoom;
+  for (let i = 0; i < dinosaurs.length; i++) {
+    for (let room of rooms) {
+      if (dinosaurName === dinosaurs[i].name) {
+        objId.push(dinosaurs[i].dinosaurId);
+      }
+      if (room.dinosaurs.includes(objId[0])) {
+        dinoRoom = room.name
+        return dinoRoom;
+      }
+    }
+  } if (objId[0] === undefined){
+    return `Dinosaur with name '${dinosaurName}' cannot be found.`;
+  }else {
+    return `Dinosaur with name '${dinosaurName}' cannot be found in any rooms.`
+  }
+}
```

Comment:
Creative approach! Pushing them into a new array is a creative approach—it shows deep thought and experimentation. Keep it up!

Another readability principle to be conscious of: an `if` on the same line as the previous `if`'s closing bracket is _very_ easily misinterpreted as being logically connected, as with `else if` chains. Try to make your `if` placement reflect its job—form follows function.'